pmm-app/tests/QAN/filters_test.js

```diff
@@ -1,127 +1,222 @@
-  qanPage.waitForNewQANPageLoaded();
+Scenario('PMM-T123 - Verify User is able to search Env & Cluster @not-pr-pipeline @qan', async (I, qanPage, qanActions) => {
+  const filters = [
+    'ps-dev',
+    'ps-dev-cluster',
+    'pgsql-repl1',
+  ];
+
+  qanActions.waitForNewQANPageLoaded();
+  I.waitForElement(qanPage.fields.filterBy, 30);
+  const countBefore = await qanActions.getCountOfItems();
+
+  for (i = 0; i < filters.length; i++) {
+    qanActions.applyFilterNewQAN(filters[i]);
+    const countAfter = await qanActions.getCountOfItems();
+
+    await qanActions.verifyChangedCount(countBefore, countAfter);
+    qanActions.applyFilterNewQAN(filters[i]);
+  }
+});
+
+Scenario('Check All Filter Groups Exists in the Filter Section @not-pr-pipeline @qan', async (I, qanPage, qanActions) => {
+  qanActions.waitForNewQANPageLoaded();
   for (i = 0; i < qanPage.filterGroups.length; i++) {
     I.fillField(qanPage.fields.filterBy, qanPage.filterGroups[i]);
     I.waitForVisible(qanPage.filterSectionLocator(qanPage.filterGroups[i]), 30);
     I.seeElement(qanPage.filterSectionLocator(qanPage.filterGroups[i]));
     I.clearField(qanPage.fields.filterBy);
   }
 });
+
+Scenario('PMM-T128 - Verify pagination works correctly @not-pr-pipeline @qan', async (I, qanPage, qanActions) => {
+  qanActions.waitForNewQANPageLoaded();
+  qanActions.verifySelectedCountPerPage('25 / page');
+  const countOfItems = await qanActions.getCountOfItems();
+
+  console.log(`Count of Items ${countOfItems}`);
+  if (countOfItems <= 50) {
+    I.seeAttributesOnElements(qanPage.fields.previousPage, { 'aria-disabled': 'true' });
+    I.click(qanPage.fields.nextPage);
+    qanActions.verifyActiveItem(2);
+    await qanActions.verifyCount(`26-${countOfItems}`);
+  } else if (countOfItems <= 100 && countOfItems > 50) {
+    I.seeAttributesOnElements(qanPage.fields.previousPage, { 'aria-disabled': 'true' });
+    I.click(qanPage.fields.nextPage);
+    qanActions.verifyActiveItem(2);
+    await qanActions.verifyCount('26-50');
+    qanActions.verifyActiveItem(1);
+    await qanActions.verifyCount('1-25');
+  } else {
+    I.seeAttributesOnElements(qanPage.fields.previousPage, { 'aria-disabled': 'true' });
+    I.click(qanPage.fields.nextPage);
+    qanActions.verifyActiveItem(2);
+    await qanActions.verifyCount('26-50');
+    I.click(qanPage.fields.previousPage);
+    qanActions.verifyActiveItem(1);
+    await qanActions.verifyCount('1-25');
+    I.seeAttributesOnElements(qanPage.fields.previousPage, { 'aria-disabled': 'true' });
+    I.click(qanPage.fields.ellipsisButton);
+    qanActions.verifyActiveItem(6);
+    await qanActions.verifyCount('126-150');
+    I.click(qanPage.fields.ellipsisButton);
+    qanActions.verifyActiveItem(1);
+    await qanActions.verifyCount('1-25');
+    qanActions.selectPage(3);
+    qanActions.verifyActiveItem(3);
+    await qanActions.verifyCount('51-75');
```

Comment:
I guess it won't work properly, because we're not checking that we have enough elements ( 75 )

Reply:
you are right, I realized it while handling other cases with 25, 50 checks, it worked now because we have a different job for QAN data to be loaded with enough wait to load those distinct queries from all instances hence it worked, but to handle this check completely it would need more "logic".